Source/CarthageKit/Git.swift

```diff
@@ -524,21 +571,33 @@ public func resolveReferenceInRepository(_ repositoryFileURL: URL, _ reference:
 	return ensureDirectoryExistsAtURL(repositoryFileURL)
 		.then(launchGitTask([ "rev-parse", "\(reference)^{object}" ], repositoryFileURL: repositoryFileURL))
 		.map { string in string.trimmingCharacters(in: .whitespacesAndNewlines) }
-		.mapError { error in CarthageError.repositoryCheckoutFailed(workingDirectoryURL: repositoryFileURL, reason: "No object named \"\(reference)\" exists", underlyingError: error as NSError) }
+		.mapError { error in
+			CarthageError.repositoryCheckoutFailed(
```

Comment:
Thanks, I'll try to come up with a custom rule for these returns as they appear quite often in your review. That sounds to me like it's a rule, so we should automate this.